server/src/main/java/org/kaazing/gateway/server/config/parse/translate/June2016Validator.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2007-2016, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaazing.gateway.server.config.parse.translate;
+
+import org.kaazing.gateway.server.config.parse.translate.june2016.FindMatchingAuthConstraint;
+import org.kaazing.gateway.server.config.parse.translate.june2016.RemoveRealmVisitor;
+
+public class June2016Validator extends GatewayConfigTranslatorPipeline {
+
+    public June2016Validator() {
+        super();
+        
+        // Ensures that each realm is accompanied by an authorization constraint, otherwise
+        // it should not exist.
+        addTranslator(new FindMatchingAuthConstraint());
```

Comment:
This should be done in the translation from old to new.  Not in the validation of the new